frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useField, useFormikContext, FormikValues } from 'formik';
+import BuilderImageCard from '@console/dev-console/src/components/import/builder/BuilderImageCard';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import './EventSourcesSelector.scss';
+
+interface EventSourcesSelectorProps {
+  eventSourceList: any;
+}
+
+const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSourceList }) => {
+  const [selected] = useField('type');
+  const { values, setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
+  const eventSourceCount = _.keys(eventSourceList).length;
+
+  const handleImageChange = React.useCallback(
+    (value: string) => {
+      setFieldValue('type', value);
+      setFieldTouched('type', true);
+      validateForm();
+    },
+    [setFieldValue, setFieldTouched, validateForm],
+  );
+
+  React.useEffect(() => {
+    if (!selected.value && eventSourceCount === 1) {
+      const image = _.find(eventSourceList);
+      handleImageChange(image.name);
+    }
+    if (!selected.value && values.type) {
+      handleImageChange(values.type);
+    }
+  }, [eventSourceCount, eventSourceList, handleImageChange, selected.value, values.type]);
+
+  if (eventSourceCount === 1) {
+    return null;
+  }
```

Comment:
Not sure if this whole logic is needed for this use case. It was necesarry to support both git import form and s2i form where there was a possibility that the builder image would come pre selected. If that is not the case here then we should remove the unnecesarry code here.

Reply:
yeah missed it , we just need `handleImageChange`